src/pyk/kbuild/project.py

```diff
@@ -61,103 +59,32 @@ class Target:
     name: str  # TODO Maybe remove name and store in project as Dict
 
     main_file: Path  # relative to source folder
-    backend: KompileBackend
-    main_module: str | None
-    syntax_module: str | None
-    md_selector: str | None
-    hook_namespaces: tuple[str, ...] | None
-    emit_json: bool | None
-    gen_bison_parser: bool | None
-    gen_glr_bison_parser: bool | None
-    bison_parser_library: bool | None
-    # LLVM backend
-    opt_level: int | None
-    ccopts: tuple[str, ...] | None
-    no_llvm_kompile: bool | None
-    enable_search: bool | None
-    enable_llvm_debug: bool | None
-    llvm_kompile_type: LLVMKompileType | None
-    llvm_kompile_output: str | None
-    # Haskell backend
-    concrete_rules: tuple[str, ...] | None
-    haskell_binary: bool | None
+    args: Mapping[str, Any]
 
     def __init__(
         self,
         *,
         name: str,
         main_file: str | Path,
-        backend: KompileBackend,
-        main_module: str | None = None,
-        syntax_module: str | None = None,
-        md_selector: str | None = None,
-        hook_namespaces: Iterable[str] | None = None,
-        emit_json: bool | None = None,
-        gen_bison_parser: bool | None = None,
-        gen_glr_bison_parser: bool | None = None,
-        bison_parser_library: bool | None = None,
-        opt_level: int | None = None,
-        ccopts: Iterable[str] | None = None,
-        no_llvm_kompile: bool | None = None,
-        enable_search: bool | None = None,
-        enable_llvm_debug: bool | None = None,
-        llvm_kompile_type: LLVMKompileType | None = None,
-        llvm_kompile_output: str | None = None,
-        concrete_rules: Iterable[str] | None = None,
-        haskell_binary: bool | None = None,
+        **kwargs: Any,
```

Comment:
Since this is instantiated automatically I think it's simpler this way:

```suggestion
        args: Mapping[str, Any],
```